src/MarkdownTextInput.web.tsx

```diff
@@ -369,20 +371,54 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
             text = parseText(divRef.current, changedText, processedMarkdownStyle).text;
         }
 
+        const normalizedText = normalizeValue(text);
+
         if (pasteRef?.current) {
           pasteRef.current = false;
           updateSelection(e);
         }
         updateTextColor(divRef.current, text);
 
         if (onChange) {
-          const event = e as unknown as NativeSyntheticEvent<any>;
+          const event = e as unknown as NativeSyntheticEvent<{
+            count: number;
+            before: number;
+            start: number;
+          }>;
           setEventProps(event);
+
+          const newSelection = CursorUtils.getCurrentCursorPosition(divRef.current) ?? {start: 0, end: 0};
```

Comment:
We can skip this step by getting `cursorPosition` from the result of the `parseText` function that is in the switch above